app/src/main/kotlin/org/eurofurence/connavigator/ui/fragments/EventItemFragment.kt

```diff
@@ -192,15 +199,14 @@ class EventItemFragment : Fragment(), HasDb, AnkoLogger {
     /**
      * Changes the FAB based on if the current event is liked or not
      */
-    private fun changeFabIcon() {
-        (eventId in db.faves).let {
-            if (it != ui.favoriteButton.tag) {
-                ui.favoriteButton.tag = it
-                if (it)
-                    ui.favoriteButton.setImageDrawable(context?.createFADrawable(R.string.fa_heart_solid))
-                else
-                    ui.favoriteButton.setImageDrawable(context?.createFADrawable(R.string.fa_heart))
-            }
+    private fun setFabIconState(isFavorite: Boolean) {
+        info("Updating icon of FAB for $eventId")
+        if (isFavorite) {
+            ui.favoriteButton.setImageDrawable(context?.createFADrawable(R.string.fa_heartbeat_solid))
```

Comment:
imo the heartbeat icon is a bad choice for this. can you get an empty heart for a not favorited state, and a solid heart for a favorited state?

Reply:
Valid point! Just found out how to do that with the SDK. Done. (: